code/datums/elements/dusts_on_leaving_area.dm

```diff
@@ -1,28 +1,21 @@
 /datum/element/dusts_on_leaving_area
 	element_flags = ELEMENT_DETACH | ELEMENT_BESPOKE
 	id_arg_index = 2
-	var/list/attached_mobs = list()
 	var/list/area_types = list()
 
 /datum/element/dusts_on_leaving_area/Attach(datum/target,types)
 	. = ..()
 	if(!ismob(target))
 		return ELEMENT_INCOMPATIBLE
-	attached_mobs += target
 	area_types = types
-	START_PROCESSING(SSprocessing,src)
+	RegisterSignal(target,COMSIG_AREA_ENTERED,.proc/check_dust)
 
 /datum/element/dusts_on_leaving_area/Detach(mob/M)
 	. = ..()
-	if(M in attached_mobs)
-		attached_mobs -= M
-	if(!attached_mobs.len)
-		STOP_PROCESSING(SSprocessing,src)
+	UnregisterSignal(M,COMSIG_AREA_ENTERED)
```

Comment:
```suggestion
	UnregisterSignal(M, COMSIG_ENTER_AREA)
```

Reply:
this is correct but unfortunately COMSIG_ENTER_AREA also has the *area* as the argument, which makes it a mite harder to work with.

Reply:
actually no i have a solution for this

Reply:
You are reading it wrong, COMSIG_ENTER_AREA is the signal sent to the mob, COMSIG_AREA_ENTERED is sent to the area.